test/unit/DocumentRetriever.js

```diff
@@ -1,5 +1,6 @@
 const {expect} = require("chai");
-const DocumentRetriever = require("../../dist/DocumentRetriever");
+const rewire = require("rewire");
+const DocumentRetriever = rewire("../../dist/DocumentRetriever");
```

Comment:
I think we should remove the dependency on `rewire` and move the necessary functions into the `utils` folder.